tests/unit_node/_fs/_fs_watch_test.ts

```diff
@@ -52,3 +53,46 @@ Deno.test({
     watcher.unref();
   },
 });
+
+Deno.test({
+  name: "node [fs/promises] watch should return async iterable",
+  sanitizeOps: false,
+  sanitizeResources: false,
+  async fn() {
+    const file = Deno.makeTempFileSync();
+    Deno.writeTextFileSync(file, "foo");
+
+    const result: { eventType: string; filename: string | null }[] = [];
+
+    const controller = new AbortController();
+    const watcher = watchPromise(file, {
+      // Node types resolved by the LSP clash with ours
+      // deno-lint-ignore no-explicit-any
```

Comment:
is it deno LSP? should we open an issue for this?

Reply:
yeah the LSP doesn't work well for me in the deno repository. It pulls in `@types/node@18.x` which is a bit outdated. In node the definition of `AbortSignal` is slightly different than the one in Deno.